libr/anal/bb.c

```diff
@@ -31,10 +31,13 @@ static bool bb_from_offset_first_cb(RAnalBlock *block, void *user) {
 }
 
 R_API RAnalBlock *r_anal_bb_from_offset(RAnal *anal, ut64 off) {
-	if (anal->opt.jmpmid && anal->cur->jmpmid) {
-		BBFromOffsetJmpmidCtx ctx = { off, NULL };
-		r_anal_blocks_foreach_in (anal, off, bb_from_offset_jmpmid_cb, &ctx);
-		return ctx.ret;
+	if (anal->opt.jmpmid) {
+		const int align = r_anal_archinfo (anal, R_ANAL_ARCHINFO_ALIGN);
+		if (align <= 1 || !(off % align)) {
```

Comment:
what about adding a function like: `if (r_anal_is_aligned (anal, off)) {` and reuse it in every call.

Reply:
:thinking: